docs/src/hugo/content/gzip.md

```diff
@@ -0,0 +1,74 @@
+---
+menu: main
+title: GZip Compression
+weight: 124
+---
+
+Http4s provides [Middleware], named `GZip`, for allowing for the compression of the `Response`
+body using GZip.
+
+Examples in this document have the following dependencies.
+
+```scala
+libraryDependencies ++= Seq(
+  "org.http4s" %% "http4s-dsl" % http4sVersion,
+  "org.http4s" %% "http4s-server" % http4sVersion
+)
+```
+
+And we need some imports.
+
+```tut:silent
+import org.http4s._
+import org.http4s.dsl._
+```
+
+Let's start by making a simple service that returns a (relatively) large string
+in it's body. We'll use `as[String]` to examine the body. The body is returned
+as a `Task[String]`, so we need to call `run` to get the string itself.
+
+```tut:book
+val service = HttpService {
+  case _ =>
+    Ok("I repeat myself when I'm under stress. " * 3)
+}
+
+val request = Request(Method.GET, uri("/"))
+
+val response = service(request).run
+val body = response.as[String].run
+body.length
```

Comment:
Maybe we want to get people out of the habit of `run`ning before the end of the world?

```
for {
  resp <- service(request.run)
  body <- resp.as[String]
} yield body.length
```